README.md

# Extra test for student 18/3/22
# IBI1_2021-22


Practical7/History.txt
Question 1 : solution line number in covid.py 7
`covid_data.head(5)`       What does the number 5 do? Change it to find out!
Conclusion: to read the head 5 line of the dataframe

Question 2 : solution line number in covid.py 8
What type are the data points in covid_data? What are the columns called? How many rows are there?
Conclusion:the first two columns are 'object' the last four columns are 'int64'; the name of columns  are 'data', 'location', 'new_cases', 'new_deaths', 'total_cases', 'tatal_deaths'; there are 7996 rows in the dataframe 

Question 3 : solution line number in covid.py 9
What was the mean of new cases (across all rows of the dataframe)? What is the standard deviation? What is the range of total cases (minimum and maximum)?
Conclusion: the mean of new cases is 194.546773; the standard deviation is 2083.395028; minimum and maximum are -9.000000 and 65162.000000

Question 4 : solution line number in covid.py 14 to 15
How would you show the second column from every 100th row from the first 1000 rows
Conclusion:   a = covid_data.iloc[100:1001:100,2]      print(a)

Question 5 : solution line number in covid.py 16-17
What happens if my_columns is shorter than the number of columns of your data frame? What happens if it’s longer?
Conclusion:    shorter:IndexError: Boolean index has wrong length: 5 instead of 6
               longer:IndexError: Boolean index has wrong length: 7 instead of 6

Question 6 : solution line number in covid.py 20-33
how do you find every row where the location is “Afghanistan”?
Conclusion:

Question 7 : solution line number in covid.py 36-45
For instance, what can we learn from looking at new cases and deaths on the last day for which data was recorded, 31 March 2020?
Conclusion:

Question 8 : solution line number in covid.py 48-55
Use numpy to compute the mean new cases and new deaths on this date. Are they similar or different? What does that tell you? What is the proportion of new deaths as a proportion
of the new cases on 31 March
Conclusion: mean:new_cases     640.461538      new_deaths     37.928205 they are different. the proportion of new deaths as a proportion is 0.05922011370005605

Question 9 : solution line number in covid.py 
Plot the new cases and new deaths on 31 March as two separate box plots
Conclusion:

Question 6 : solution line number in covid.py 

Conclusion:

Practical7/Practical7.py
import os
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
os.chdir("C:/Users/33244/Desktop/IBI1_2022-23/IBI1_2022-23/Practical7")
covid_data = pd.read_csv("full_data.csv")
covid_data.head(5) #read the head 5 line of the dataframe
covid_data.info()
print(covid_data.describe(include='all'))
covid_data.iloc[0,1]
covid_data.iloc[2,0:5]
covid_data.iloc[0:2,:]
covid_data.iloc[0:10:2,0:5]
a = covid_data.iloc[100:1001:100,2]
print(a)
my_columns = [True, True, False, True, False, False]
covid_data.iloc[0:3,my_columns]
covid_data.loc[2:4,"date"]
covid_data.loc[0:81,"total_cases"]
#creat a variable to calculate the number of loop
#use a while loop to test the place
 #add a boolean value to the variable
 #change the index
#print the Afghanistan's total_cases
Afghanistan_rows = []
i = 0
while i != 7996:
    if covid_data.loc[i,"location"] == "Afghanistan":
       Afghanistan_rows.append(i)
    i = i + 1

print(Afghanistan_rows)
covid_data.iloc[Afghanistan_rows,5]


new_data_rows = []
i = 0
while i != 7996:
    if covid_data.loc[i,"date"] == "2020-03-31":
       new_data_rows.append(i)
    i = i + 1

new_data_columns = [False, True, True, True, False, False]
new_data = covid_data.iloc[new_data_rows,new_data_columns]
print(new_data)


mean = new_data[['new_cases','new_deaths']].mean()
print(mean)
new_cases = new_data.loc[:, 'new_cases']
mean_new_cases = np.mean(new_cases)
new_deaths = new_data.loc[:, 'new_deaths']
mean_new_deaths = np.mean(new_deaths)
result = mean_new_deaths / mean_new_cases
print (result)



# Create two separate box plots using matplotlib
boxprops = dict(linestyle='--', linewidth=2, color='black')
fig, axs = plt.subplots(1, 2, figsize=(10, 5))
axs[0].boxplot(new_cases,boxprops=boxprops)
axs[0].set_title('New Cases on March 31')
axs[1].boxplot(new_deaths,boxprops=boxprops)
axs[1].set_title('New Deaths on March 31')
plt.show()




Algeria = []
i = 0
while i != 7996:
    if covid_data.loc[i,"location"] == "Algeria":
       Algeria.append(i)
    i = i + 1

print(Algeria)
world_data = covid_data.loc[Algeria,['total_cases','total_deaths']]
world_dates = covid_data.loc[Algeria,'date']
plt.plot(world_dates, world_data['total_cases'], 'bo')
plt.plot(world_dates, world_data['total_deaths'], 'rs')
plt.xticks(world_dates.iloc[0:len(world_dates):4],rotation=-90)
plt.show()




Austria = []
i = 0
while i != 7996:
    if covid_data.loc[i,"location"] == "Austria":
       Austria.append(i)
    i = i + 1

print(Austria)
Austria_data = covid_data.loc[Austria,['total_cases','total_deaths']]
Austria_dates = covid_data.loc[Austria,'date']
plt.plot(Austria_dates, Austria_data['total_cases'], 'ro')
plt.plot(Austria_dates, Austria_data['total_deaths'], 'bo')
plt.xticks(Austria_dates.iloc[0:len(Austria_dates):4],rotation=-90)
plt.title('Total cases and deaths in Austria')
plt.xlabel('Date')
plt.ylabel('number')
plt.show()

UK = []
i = 0
while i != 7996:
    if covid_data.loc[i,"location"] == "United Kingdom":
       UK.append(i)
    i = i + 1

UK_data = covid_data.loc[UK,['new_cases','total_cases']]
UK_dates = covid_data.loc[UK,'date']
plt.plot(UK_dates, UK_data['new_cases'], 'ro')
plt.plot(UK_dates, UK_data['total_cases'], 'bo')
plt.xticks(UK_dates.iloc[0:len(UK_dates):4],rotation=-90)
plt.title('Trends in total and new cases in the United Kingdom')
plt.xlabel('Date')
plt.ylabel('number')
plt.show()





